### ISSUE TITLE
Different .show behaviors in notebooks for version 0.1.5 and 0.1.6

### ISSUE THREAD

**Issue by shyamsn97** [Eventual-Inc/Daft#1075]
**Describe the bug**
Calling `.show` on a daft dataframe has different behaviors in 0.1.5 vs 0.1.6

**To Reproduce**
In a notebook, run this code w/ getdaft==0.1.5 and 0.1.6:

```python
import daft
from daft import DataType
from PIL import Image
import requests
import numpy as np

images = [
    "https://raw.githubusercontent.com/Purukitto/pokemon-data.json/master/images/pokedex/hires/001.png",
]

df = daft.from_pydict({"images":images, "initial_array":np.zeros((1,512))})

@daft.udf(return_dtype=DataType.python())
def download_images(images):
    return [Image.open(requests.get(im, stream=True).raw).convert("RGB") for im in images.to_pylist()]

@daft.udf(return_dtype=DataType.python())
def arrays(column):
    return np.zeros((1, 512))

df = df.with_column("downloaded_images", download_images(daft.col('images'))).collect()
df = df.with_column("embeddings", arrays(daft.col("images"))).collect()

df.show(1)
```
In the code above I'm initializing a dataframe with an image url and a dummy array. Then I'm creating a column with a PIL Image that is downloaded and adding another column with a dummy array for the "embeddings"

Version 0.1.5:

![Screenshot from 2023-06-21 12-11-01](https://github.com/Eventual-Inc/Daft/assets/18158178/e5aea028-d8a1-47fb-b2f4-25ac3df06f03)

Version 0.1.6:
![Screenshot from 2023-06-21 12-10-38](https://github.com/Eventual-Inc/Daft/assets/18158178/02838e2b-bae5-4fb5-9371-c448766382c5)

Not sure if this is on my end w/ respect to jupyter versions / other python pkg versions. Let me know if you need any additional context, thanks!

**Additional context**
jupyter-lab==4.0.2
python==3.9

**Comment by jaychia** [comment]
Indeed! You actually should be able to get the displays working by using our Daft native image types instead.

I modified your code to take advantage of the new image types - using `.url.download().image.decode()`

```python
import daft
from daft import DataType
from PIL import Image
import requests
import numpy as np

images = [
    "https://raw.githubusercontent.com/Purukitto/pokemon-data.json/master/images/pokedex/hires/001.png",
]

df = daft.from_pydict({"images":images, "initial_array":np.zeros((1,512))})

@daft.udf(return_dtype=DataType.python())
def arrays(column):
    return np.zeros((1, 512))

df = df.with_column("downloaded_images", daft.col('images').url.download().image.decode()).collect()
df = df.with_column("embeddings", arrays(daft.col("images"))).collect()

df.show(1)
```

Regardless though, we should re-enable this display feature for PIL Python types! I'll work on re-enabling this feature so that if you have a `python()` type column with PIL images it will detect the object and display it correctly.

**Comment by shyamsn97** [comment]
awesome! Thanks for the explanation :)

**Comment by jaychia** [comment]
This should be fixed by #1078!
